### ISSUE TITLE
Configurable raw object editor control

### ISSUE THREAD

**Issue by Ludevik** [storybookjs/storybook#15900]
**Is your feature request related to a problem? Please describe**
Controls addon for objects displays json using `react-editable-json-tree`. It is set to not use `raw` editor by default. This causes a bit of a problem for automated cypress tests as it is easier to interact with simple textarea than with json tree component.

**Describe the solution you'd like**
We would like to have configurable object editor through `argTypes`, something like:
`control: { type: 'object', useRaw: true }`

**Describe alternatives you've considered**
I didn't found any alternative to this.

**Are you able to assist to bring the feature to reality?**
maybe yes :)

**Additional context**
![image](https://user-images.githubusercontent.com/881672/130496392-3abbb037-c2a5-4d37-93e4-1f5738fe47b9.png)

**Comment by shilman** [comment]
@ghengeveld WDYT?

**Comment by Ludevik** [comment]
@shilman writing them. the test changes the value through control and then expects something. 

I guess we could also use URL parameters like this:
`/story/dialog&args=data.title:Some+Test`

Problem is that some content is not safe to be used in params, e.g.: `{ "title": "Some.Test" }` and it is stripped automatically when using control, what about such cases?

**Comment by leepowelldev** [comment]
I too would like to see the option of configuring 'raw' mode. The free text version is easy to error for end users ... the key/value view would be great to have as a default.

**Comment by shilman** [comment]
Yeah I'm not exactly sure why the raw mode is the default. @ghengeveld WDYT?

**Comment by Ludevik** [comment]
Hm, i am on version 6.3.8 and i have key/value view by default instead of raw/text mode. I am using `@storybook/angular`.

**Comment by shilman** [comment]
Chatted with @ghengeveld and key/value is the default unless the object is `undefined` in which case it switches to the raw mode.

**Comment by leepowelldev** [comment]
I'm seeing raw mode when the default value is and empty object. Would be great to have this configurable.

**Comment by tripolskypetr** [comment]
+1 I was looking for solution of this problem but it looks like still not implemented

**Comment by tripolskypetr** [comment]
I think for now monkey-patch the html with [auto-clicker script](https://github.com/react-declarative/react-declarative-storybook/blob/master/template/object-auto-expand.js) will be the simplest. If you want, you can use [this project as a sample](https://github.com/react-declarative/react-declarative-storybook)

```
const observer = new MutationObserver(() => {
    const expandButtons = [...document.querySelectorAll('button[type=button] > span')]
        .filter((el) => el.innerHTML === 'RAW')
        .map((el) => el.closest('button'));
    for (const button of expandButtons) {
        if (!button.classList.contains(EXPANDED_CLASS)) {
            button.classList.add(EXPANDED_CLASS);
            button.click();
        }
    }
});
observer.observe(document.body, {
    childList: true,
    subtree: true,
});
```

**Comment by EliezerB123** [comment]
> I think for now monkey-patch the html with [auto-clicker script](https://github.com/react-declarative/react-declarative-storybook/blob/master/template/object-auto-expand.js) will be the simplest. If you want, you can use [this project as a sample](https://github.com/react-declarative/react-declarative-storybook)
> 
> ```
> const observer = new MutationObserver(() => {
>     const expandButtons = [...document.querySelectorAll('button[type=button] > span')]
>         .filter((el) => el.innerHTML === 'RAW')
>         .map((el) => el.closest('button'));
>     for (const button of expandButtons) {
>         if (!button.classList.contains(EXPANDED_CLASS)) {
>             button.classList.add(EXPANDED_CLASS);
>             button.click();
>         }
>     }
> });
> observer.observe(document.body, {
>     childList: true,
>     subtree: true,
> });
> ```
> 
> **P.S.** Check webpack config in [`config/main`](https://github.com/react-declarative/react-declarative-storybook/blob/1be4c67c3417606d7d57b6eea3695747e7d8c929/config/main.js). After production build, the patch will be installed automatically

@tripolskypetr Wow. It's crazy and I love it. Very interesting method of creating a workaround. Thank you!

**Comment by mokone91** [comment]
it would be nice to have configuration for this case

**Comment by LeonxLJX** [comment]
I'd like to take this one.

I've read through the report. Before changing anything I'd like to reproduce it against current `main` first — my reasoning is that a fix anchored to the actual observed behavior is much easier to review than one inferred from the report alone, especially if the surrounding code has moved since this was filed.

Plan: reproduce and confirm the root cause, then make the smallest change that addresses it (plus a regression test if the repo has coverage for this area). If the root cause turns out to live somewhere other than where the report points, I'll report back here before forcing a change.

Anything already known about this that I should read first?

(claiming via @LeonxLJX)
